### PR TITLE
Update term timeout

### PR DESCRIPTION
**Pull request by Parsifal22**
I rewrote the [_on_term_timeout(self)](https://github.com/Parsifal22/cyraft/blob/9504d06092f9e35cdc896265e8415ac916dd9aec/cyraft/node.py#L717) method in much the same way as Pavel did in the [libcyphal project](https://github.com/OpenCyphal-Garage/libcyphal/blob/dcc3a4de237b7482e04543d2393c3a9385685312/libuavcan/include/uavcan/protocol/dynamic_node_id_server/distributed/raft_core.hpp#L260-L316). 
In a nutshell. In the rewritten version, the method is called more often and sends a heartbeat to only one follower. If the follower fails to complete the task within the allotted time (if it is not in the cluster or simply stalls), then with the next call to _on_election_timeout(self) it cancels the heartbeat request and sends a heartbeat to another follower.

### CODE REVIEW

**Comment by Parsifal22** [comment]
I've added a new [test case](https://github.com/Parsifal22/cyraft/blob/295629400841e59b99cae7e3e151f08d7c61ca0a/tests/raft_leader_election.py#L419C1-L505C27) for rewritten methode `_on_term_timeout(self)`
